extensions/tasks-parent/tasks-client/src/main/java/io/resys/thena/tasks/client/spi/visitors/HistoryVisitor.java

```diff
@@ -0,0 +1,38 @@
+package io.resys.thena.tasks.client.spi.visitors;
+
+import io.resys.thena.docdb.api.models.Objects.BlobHistory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class HistoryVisitor {
+
+  private final List<BlobHistory> collected = new ArrayList<>();
+  private BlobHistory previous = null;
+
+  public HistoryVisitor visitBlobHistory(List<BlobHistory> history) {
+    history.forEach(this::visitBlobHistory);
+    return this;
+  }
+
+  private void visitBlobHistory(BlobHistory history) {
+    if(previous == null) {
+      collected.add(history);
+      previous = history;
+      return;
+    }
+
```

Comment:
You don't need to do anything about this -- just a note:

.equals() checks for memory allocation.  But this behaviour can be overridden. In our case, the blob object is immutable, and overrides the equality check in the implementation. But, if this were not an immutable object, you'd have to check for `blob.getId()`.